tests/jobs/test_cherrypicker_data_job.py

```diff
@@ -86,24 +79,35 @@ def mongo_collection(mongo_database):
     yield config, get_mongo_collection(mongo_database, COLLECTION_CHERRYPICK_TEST_DATA)
 
 
-@pytest.fixture
-def mock_stack():
-    with ExitStack() as stack:
-        datetime_mock = stack.enter_context(patch("crawler.jobs.cherrypicker_test_data.datetime"))
-        datetime_mock.utcnow.return_value = mocked_utc_now
-        stack.enter_context(patch("crawler.jobs.cherrypicker_test_data.create_barcodes", return_value=created_barcodes))
-        stack.enter_context(patch("crawler.jobs.cherrypicker_test_data.create_csv_rows", return_value=created_csv_rows))
-        stack.enter_context(patch("crawler.jobs.cherrypicker_test_data.write_plates_file"))
-        stack.enter_context(patch("crawler.jobs.cherrypicker_test_data.run_crawler"))
-        stack.enter_context(
-            patch("crawler.jobs.cherrypicker_test_data.create_barcode_meta", return_value=created_barcode_metadata)
-        )
+@pytest.fixture(autouse=True)
+def ctpd_processor_class():
+    with patch("crawler.jobs.cherrypicker_test_data.CPTDProcessor") as cptd_processor_class:
+        yield cptd_processor_class
+
+
+@pytest.fixture(autouse=True)
+def create_barcode_meta_method():
+    with patch("crawler.jobs.cherrypicker_test_data.create_barcode_meta") as method:
+        method.return_value = CREATED_BARCODE_METADATA
+        yield method
+
 
-        yield stack
+@pytest.fixture(autouse=True)
```

Comment:
I haven't used autouse before. Have we checked that this fixture autouse does not create the fixture for tests outside of this file?

Reply:
I've tested how the scope works.  If I move this fixture's definition to another test file, leaving it otherwise unchanged, the tests in this file fail.  It is only applicable to the file it appears in.